### DISCUSSION TITLE
Is this a type narrowing bug?

### DISCUSSION THREAD

**Discussion by robert-bn** [microsoft/pyright#8939] [Q&A] [answered]
I'm not sure if this is intended behaviour or not:

When checking referential equality to `Ellipsis` (`...`) the narrowed type according to Pyright 1.1.379 CLI is `None`, rather than `EllipsisType`.

```python
def example(x: object):
    if x is ...:
        reveal_type(x)  # Pylance: Type of "x" is "None"
```

**Comment by erictraut** [comment, marked as answer]
Yes, that's a bug. Thanks for reporting it. This will be addressed in the next release.
